### PR TITLE
[PBW-5708] verify saved caption lang availability

### PR DESCRIPTION
**Pull request by mistermendez**

### CODE REVIEW

File: js/controller.js [hunk 1]
```diff
@@ -1222,6 +1222,10 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
     },
 
     setClosedCaptionsLanguage: function(){
+      //if saved language not in available languages, set to first available language
+      if (this.state.closedCaptionOptions.language == null || !_.contains(this.state.closedCaptionOptions.availableLanguages.languages, this.state.closedCaptionOptions.language)) {
+        this.state.closedCaptionOptions.language = this.state.closedCaptionOptions.availableLanguages.languages[0];
```
Comment (mityaha) [inline js/controller.js]: Is it always a sure thing that array this.state.closedCaptionOptions.availableLanguages.languages is not empty?  Don't you also want to check whether it is not empty?

Comment (mistermendez) [inline js/controller.js]: No, the contains() function would just return false.

**Comment by mistermendez** [comment]
@mityaha does this also fix PBW-5709 ?

**Comment by mityaha** [comment]
I haven't built with this change, I'll try.

File: js/controller.js (hunk 1 above)
Comment (aldigun) [inline js/controller.js]: Shouldn't we default to the language set in skin.json first, then fallback to first language.

Comment (aldigun) [inline js/controller.js]: Oh nvm, during controller creation, language selection is already extended to contain the skin.json setting.

Comment (mistermendez) [inline js/controller.js]: yup done here - https://github.com/ooyala/html5-skin/pull/618/files#diff-ceb739c986db75454cd0d5c8f22f2928L228

Comment (aldigun) [inline js/controller.js]: LGTM.

**Comment by aldigun** [comment]
This defintely fixes PBW-5708 and PBW-5712

**Pull request merged**